src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1320,21 +1321,35 @@ synchronized public void onChangePredicate(Chat chat, String predicateName, Stri
     for (Session s : sessions.values()) {
       if (s.chat == chat) {
         // found session saving predicates
-        invoke("publishChangePredicate", s, chat, predicateName, result);
+        invoke("publishPredicate", s, predicateName, result);
         s.savePredicates();
         return;
       }
     }
     error("could not find session to save predicates");
   }
 
-  public PredicateEvent publishChangePredicate(Session session, Chat chat, String name, String value) {
+  /**
+   * Predicate updates are published here.  Topic (one of the most important predicate change) is also published
+   * when it changes. Session is needed to extract current user and bot this is relevant to.
+   * @param session - session where the predicate change occurred
+   * @param name - name of predicate
+   * @param value - new value of predicate
+   * @return
+   */
+  public PredicateEvent publishPredicate(Session session, String name, String value) {
     PredicateEvent event = new PredicateEvent();
     event.id = String.format("%s<->%s", session.userName, session.botInfo.name);
     event.userName = session.userName;
     event.botName = session.botInfo.name;
     event.name = name;
     event.value = value;
+    
+    if ("topic".equals(name)) {
+      invoke("publishTopic", new TopicChange(session.userName, session.botInfo.name, value, session.currentTopic));

```

Comment:
we should probably also test that the new value is actually different from the old value here, o/w we could falsely detect a change in topic, maybe?

Reply:
updated to filter out non changes